cpp/rrtmgp_conversion.h

```diff
@@ -423,11 +451,91 @@ void unflatten_idx_right(const int idx, const Kokkos::Array<int, 2>& dims, int&
 KOKKOS_INLINE_FUNCTION
 void unflatten_idx_right(const int idx, const Kokkos::Array<int, 3>& dims, int& i, int& j, int& k)
 {
-  i = (idx / dims[2]) / dims[1];
+  i = idx / (dims[2] * dims[1]);
   j = (idx / dims[2]) % dims[1];
   k =  idx % dims[2];
 }
 
+KOKKOS_INLINE_FUNCTION
```

Comment:
Some of the interesting changes are here. These are the functions that unflatten an MD idx and the macros that makes this stuff usable.